core/src/main/scala/akka/kafka/ProducerSettings.scala

```diff
@@ -349,7 +399,11 @@ class ProducerSettings[K, V] @InternalApi private[kafka] (
    * (without blocking for `enriched`).
    */
   def createKafkaProducerAsync()(implicit executionContext: ExecutionContext): Future[Producer[K, V]] =
-    enriched.map(producerFactory)
+    (producerFactoryAsync, producerFactorySync) match {
+      case (Some(asyncFactory), _) => asyncFactory(executionContext)
```

Comment:
What is the use case for the `asyncFactory`?
Why doesn't it get the enriched settings passed?

Reply:
I copied this from the `Transactional.flow` implementation that's been been removed in this PR. 
 The "async" producer factory was passed to `TransactionalProducerStage`: https://github.com/akka/alpakka-kafka/pull/952/files#diff-832423d99d7759061130b358267d97a3L101

I don't know if we need this.  We no longer require it internally.

Reply:
It seems irregular to me, remove it.

Reply:
I [re-]discovered the documentation example used to point to this overload that takes a `Future[KafkaProducer[K,V]]` that was added with the discovery PR (https://github.com/akka/alpakka-kafka/pull/952/files#diff-832205ad71b0f8522d293275289adc35L60).  It seems like the only use for this is if you were to create a `KafkaProducer` asynchronously externally from Alpakka Kafka, but I'm not sure why we need to support that.

I'll revert the documentation to showing how to share an external `KafkaProducer` synchronously.

Reply:
I guess it's because `ProducerSettings` exposes `createKafkaProducerAsync` and `createKafkaProducerCompletionStage`.  I'm using this internally for the transactions PR.  Maybe it would be useful to share to the user if they want to get an asynchronous producer for some other purpose?

Reply:
I'll hold off removing this for now.  

I squashed the commits in this branch to make rebasing easier for the transactions PR.

Reply:
Ah, thank you for digging. Yes, I added those to allow passing the enriched producer to the factory methods with a shared producer. With this solution for passing the producer factory, we don't need those.

Reply:
Ok